src/lib/y2partitioner/dialogs/btrfs_subvolume.rb

```diff
@@ -266,6 +270,201 @@ def help
         # @return [Actions::Controllers::BtrfsSubvolume]
         attr_reader :controller
       end
+
+      # Widget to set the size of BtrfsSubvolume#referenced_limit, if quota
+      # support is enabled for the Btrfs
+      class SubvolumeRferLimit < CWM::CustomWidget
+        # Constructor
+        #
+        # @param controller [Actions::Controllers::BtrfsSubvolume]
+        #   a controller collecting data for a subvolume to be created or edited
+        def initialize(controller)
+          textdomain "storage"
+          @controller = controller
+        end
+
+        # @macro seeAbstractWidget
+        def contents
+          VBox(
+            Left(check_box_widget),
+            Left(size_widget)
+          )
+        end
+
+        # @macro seeAbstractWidget
+        def help
+          _(
+            "<p><b>Limit Size</b> allows to set a quota on the referenced space of the " \
+            "subvolume. This is only possible if Btrfs quotas are enabled for this file " \
+            "system. Btrfs quotas can be enabled or disabled editing the file system " \
+            "from the Btrfs section of the Partitioner.</p>"
+          )
+        end
+
+        # Disables the widget if quotas are not enabled
+        #
+        # @see #disable_widgets
+        def init
+          disable_widgets unless quota?
+        end
+
+        # @macro seeAbstractWidget
+        def value
+          if quota? && check_box_widget.value
+            size_widget.value
+          else
+            Y2Storage::DiskSize.unlimited
+          end
+        end
+
+        # @macro seeAbstractWidget
+        def store
+          @controller.subvolume_referenced_limit = value
+        end
+
+        # @macro seeAbstractWidget
+        def validate
+          return true if value
+
+          Yast::Popup.Error(_("The size entered is invalid."))
```

Comment:
NP: I understood we prefer to use `Yast2::Popup` in the new code.

Reply:
Well, for very simple cases like this, the old API is more straightforward... and we have another 27 usages of `Yast::Popup` in the Partitioner anyways.